File: julia8024/PGS_131127.py
from collections import Counter

def solution(want, number, discount):
    answer = 0
    
    # {"banana" : 3, "apple" : 2,...} 형태의 딕셔너리 구조로 변경
    dicts = {x:y for x, y in zip(want, number)}
    
    # 열흘간의 상품 종류&수량이 원하는 상품 종류&수량과 일치하는 경우 +1
    for i in range(len(discount)):
        if dicts == Counter(discount[i:i+10]):
            answer += 1
    
    return answer

File: chaeryeon823/PGS_131127.js
// 할인 행사
// lv 2
// Counter 카운터

function Counter(array) {
  const counter = {};
  array.forEach((val) => (counter[val] = (counter[val] || 0) + 1));
  return counter;
}

function check(dictWantNumber, dis) {
  const counterDis = Counter(dis);

  const keys1 = Object.keys(dictWantNumber);
  const keys2 = Object.keys(counterDis);

  if (keys1.length !== keys2.length) return false;

  return keys1.every(
    (key) =>
      dictWantNumber[key] === counterDis[key] && counterDis.hasOwnProperty(key)
  );
}

function solution(want, number, discount) {
  let answer = 0;
  const dic = {};

  for (let i = 0; i < want.length; i++) {
    dic[want[i]] = number[i];
  }

  for (let i = 0; i <= discount.length - 10; i++) {
    answer += check(dic, discount.slice(i, i + 10)) ? 1 : 0;
  }

  return answer;
}
